contentcuration/contentcuration/viewsets/caption.py

```diff
@@ -0,0 +1,127 @@
+import logging
+
+from le_utils.constants.format_presets import AUDIO, VIDEO_HIGH_RES, VIDEO_LOW_RES
+from rest_framework import serializers
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from contentcuration.models import CaptionCue, CaptionFile, Change, File
+from contentcuration.tasks import generatecaptioncues_task
+from contentcuration.viewsets.base import BulkModelSerializer, ValuesViewset
+from contentcuration.viewsets.sync.constants import CAPTION_FILE
+from contentcuration.viewsets.sync.utils import generate_update_event
+
+
+class CaptionCueSerializer(BulkModelSerializer):
+    class Meta:
+        model = CaptionCue
+        fields = ["text", "starttime", "endtime", "caption_file_id"]
+
+    def validate(self, attrs):
+        """Check that the cue's starttime is before the endtime."""
+        attrs = super().validate(attrs)
+        if attrs["starttime"] > attrs["endtime"]:
+            raise serializers.ValidationError("The cue must finish after start.")
+        return attrs
+
+    def to_internal_value(self, data):
+        """
+        Copies the caption_file_id from the request data
+        to the internal representation before validation.
+        Without this, the caption_file_id would be lost
+        if validation fails, leading to errors.
+        """
+        caption_file_id = data.get("caption_file_id")
+        value = super().to_internal_value(data)
+
+        if "caption_file_id" not in value:
+            value["caption_file_id"] = caption_file_id
+        return value
+
+
+
+class CaptionFileSerializer(BulkModelSerializer):
+    caption_cue = CaptionCueSerializer(many=True, required=False)
+
+    class Meta:
+        model = CaptionFile
+        fields = ["id", "file_id", "language", "caption_cue"]
+
+
+class CaptionViewSet(ValuesViewset):
+    # Handles operations for the CaptionFile model.
+    queryset = CaptionFile.objects.prefetch_related("caption_cue")
+    permission_classes = [IsAuthenticated]
+    serializer_class = CaptionFileSerializer
+    values = ("id", "file_id", "language")
+
+    field_map = {
+        "file_id": "file_id",
+        "language": "language",
+    }
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        contentnode_ids = self.request.GET.get("contentnode__in")
+        file_id = self.request.GET.get("file_id")
+        language = self.request.GET.get("language")
+
+        if contentnode_ids:
+            contentnode_ids = contentnode_ids.split(",")
+            file_ids = File.objects.filter(
+                preset_id__in=[AUDIO, VIDEO_HIGH_RES, VIDEO_LOW_RES],
+                contentnode_id__in=contentnode_ids,
+            ).values_list("pk", flat=True)
+            queryset = queryset.filter(file_id__in=file_ids)
+
+        if file_id:
+            queryset = queryset.filter(file_id=file_id)
```

Comment:
Similarly to the above comment, could you also check `File.filter_edit_queryset` here

Reply:
```py
        if file_id:
            allowed_file_id = set(
                File.filter_edit_queryset(
                    File.objects.get(pk=file_id), self.request.user
                )
                .values_list("id", flat=True)
            )
            queryset = queryset.filter(file_id=allowed_file_id)
```

Please check this. 

I think there won't be comma seperated `file_ids`. (we don't actually use this query for now)